File: candyshop/candy_shop.py
# We run a Candy shop where we sell candies and lollipops
# One lollipop's price is 10$
# And it made from 5gr of sugar
# One candie's price is 20$
# And it made from 10gr of sugar
# we can raise their prices with a given percentage
#
# Create a CandyShop class
# It can store sugar and money as income. The constructor should take the amount of sugar.
# we can create lollipops and candies store them in the CandyShop's storage
# If we create a candie or lollipop the CandyShop's sugar reduce the sugar with the amount of the sweets
# We can raise the prices of all candies and lollipops with a given percentage
# We can sell candie or lollipop with a given number as amount
# If we sell sweets the income will raise with the price of the sweets and we delete it from the inventory
# We can buy sugar with a given number as amount. The price of 1000gr sugar is 100$
# If we buy sugar we can raise the CandyShop's amount of sugar and reduce the income.
# The CandyShop should be represented as string in this format:
# "Inventory: 3 candies, 2 lollipops, Income: 100, Sugar: 400gr"

candy_shop = CandyShop(300)
candy_shop.create_sweets("candy")
candy_shop.create_sweets("candy")
candy_shop.create_sweets("lollipop")
candy_shop.create_sweets("lollipop")
print(candy_shop)
# Should print out:
# Invetory: 2 candies, 2 lollipops, Income: 0, Sugar: 270gr
candy_shop.sell("candy", 1)
print(candy_shop
# Should print out:
# "Invetory: 1 candies, 2 lollipops, Income:20, Sugar: 285gr"
candy_shop.raise_prices(5)
candy_shop.sell("lollipop", 1)
print(candy_shop)
# Should print out:
# "Invetory: 1 candies, 1 lollipops, Income:35, Sugar: 285gr"
candy_shop.buy_sugar(300)
print(candy_shop)
# Should print out:
# "Invetory: 1 candies, 1 lollipops, Income:5, Sugar: 315gr"

File: src/CandyShop.java
public class CandyShop {
    int sugar;
    int money;
    int lollipops;
    int candies;
    Lollipop LOLLIPOP;
    Candies CANDY;

    public CandyShop(int sugarGramms) {
        this.sugar = sugarGramms;
        LOLLIPOP = new Lollipop();
        CANDY = new Candies();
    }
    public void createSweets(Sweets newSweets) {
        if (newSweets == CANDY) {
            candies++;
        } else {
            lollipops++;
        }
        this.sugar -= newSweets.sugar;
    }
    public void raise(int percent) {
        float percentage = 1.0f+(percent/100.0f);
        LOLLIPOP.setLollipopPrice((int) (LOLLIPOP.price * percentage));
        CANDY.setCandyPrice((int) (CANDY.price * percentage));
    }
    public void sell(Sweets sweets,int amount) {
        if (sweets == CANDY) {
            this.candies -= amount;
            this.money += CANDY.price * amount;
        } else {
            this.lollipops -= amount;
            this.money += LOLLIPOP.price * amount;
        }
    }
    public void buySugar(int sugarAmountBuy) {
        this.sugar += sugarAmountBuy;
    }
    public String printInventory() {
        String inventory = "Inventory: " + this.candies + " candies, " + this.lollipops + " lollipops, " + "Income: " + this.money + "$, " + "Sugar: "+ this.sugar;
        return inventory;
    }
}
